packages/sdk-js/src/GrowthBook.ts

```diff
@@ -106,9 +106,13 @@ export class GrowthBook<
       this._updateAllAutoExperiments();
     }
 
-    if (context.clientKey) {
+    if (context.clientKey && !context.remoteEval) {
       this._refresh({}, true, false);
     }
+
+    if (context.remoteEval && !context.userId) {
+      throw new Error("Remote eval: missing userId");
+    }
```

Comment:
In CSR applications, you might not know the userId until the app hydrates. Is there a way we can allow lazy setting this?

Reply:
Also, since we have a `setUserId` function, it's possible for someone to unset this after the constructor, so just having the check here is not sufficient.